core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt

```diff
@@ -326,11 +351,84 @@ class Sep31ServiceTest {
     // TODO: Add fee validation.
     assertEquals("100", txn.amountIn)
     assertEquals(
-      "stellar:USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN",
-      txn.amountInAsset,
+        "stellar:USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN",
+        txn.amountInAsset,
     )
     assertEquals("12500", txn.amountOut)
     assertEquals(stellarJPYC, txn.amountOutAsset)
+
+    // Test null quote failure
+    Context.get().setQuote(null)
+    assertThrows<ServerErrorException> { sep31Service.updateTxAmountsBasedOnQuote() }
+  }
+  @Test
+  fun test_getTransaction() {
+    assertThrows<BadRequestException> { sep31Service.getTransaction(null) }
+    assertThrows<BadRequestException> { sep31Service.getTransaction("") }
+
+    every { txnStore.findByTransactionId("not_found") } returns null
+    assertThrows<NotFoundException> { sep31Service.getTransaction("not_found") }
+
+    every { txnStore.findByTransactionId("found") } returns txn
+    val getTransactionResponse = sep31Service.getTransaction("found")
+    val tr = getTransactionResponse.transaction
+    assertEquals(tr.status, txn.status)
+    if (tr.statusEta != null) assertEquals(tr.statusEta, txn.statusEta)
+    assertEquals(tr.amountIn, txn.amountIn)
+    assertEquals(tr.amountInAsset, txn.amountInAsset)
+    assertEquals(tr.amountOut, txn.amountOut)
+    assertEquals(tr.amountOutAsset, txn.amountOutAsset)
+    assertEquals(tr.amountFee, txn.amountFee)
+    assertEquals(tr.amountFeeAsset, txn.amountFeeAsset)
+    assertEquals(tr.stellarMemo, txn.stellarMemo)
+    assertEquals(tr.stellarMemoType, txn.stellarMemoType)
+    assertEquals(tr.completedAt, txn.completedAt)
+    assertEquals(tr.stellarTransactionId, txn.stellarTransactionId)
+    assertEquals(tr.externalTransactionId, txn.externalTransactionId)
+    assertEquals(tr.refunded, txn.refunded)
+    assertEquals(tr.requiredInfoMessage, txn.requiredInfoMessage)
+    assertEquals(tr.requiredInfoUpdates, txn.requiredInfoUpdates)
```

Comment:
👀  Why not assert the whole object in one line? Like `assertEquals(txn, gotTxn)`. Otherwise, the test won't fail when/if we add extra fields to the object schema.

Maybe you'll need to call `tr.id = txn.id` first.

Reply:
Also, the order of the parameters seems inverted, the assertion methods usually take the expected value first and the actual value later, like in `assertEquals(wantTxn, gotTxn)`

Reply:
`tr` and `txn` are actually different types. 😢 

This is to make sure the fields of the `txn` returned from the datastore are copied correctly to the response.